nuxeo-coldstorage/src/main/java/org/nuxeo/coldstorage/helpers/ColdStorageHelper.java

```diff
@@ -220,6 +222,19 @@ public static ColdStorageContentStatus checkColdStorageContentAvailability(CoreS
         return new ColdStorageContentStatus(beingRetrieved, available);
     }
 
+    /**
+     * Retrieves the {@link DocumentModel} of the given {@code id}.
+     *
+     * @param session the core session
+     * @param id
+    the document id
+     * @return an optional containing the document model of the given {@code id} if it exists, empty otherwise
+     */
+
+    public static Optional<DocumentModel> getDocumentModelById(CoreSession session, DocumentRef documentIdRef) {
+        return Optional.of(session.getDocument(documentIdRef));
+    }
+
     protected static String getContentBlobKey(Blob coldContent) {
         String key = ((ManagedBlob) coldContent).getKey();
```

Comment:
Downcast can throw an error.
getKey methods should be provided by Blob class

Reply:
1) Change the argument type to use a new Interface that defines the `getKey` method. ManageBlob instances should implement this Interface
2) Change argument to ManageBlob. In this case. method cannt be use for other Blob types